### PR TITLE
1100354 supersearch

### PR DESCRIPTION
**Pull request by willkg**
I pulled down a fresh super search fields set from -prod earlier today. Then set up the first commit to be just like the previous PR and follow-up, but with an ES 1.4-friendly set of fields. Otherwise it's almost exactly the same.

Then I did a commit that moved the JSON file to Python.

Then two commits to clean up the resulting file.

Now the tests pass and super search works in the webapp in my local dev environment.

I didn't spend time refactoring that massive Python data structure pulling out redundant information thus making it maintainable and reducing the size of it. We can do that in a different bug after some discussion about how we want to deal with it.

I think that covers this.

To review, I suggest the following:

1. wait for Circle/Travis to be happy/unhappy
2. pull it locally and run it through paces
3. skim the changes, but that's going to be pretty hard given that there are like 5000 lines of changes in here

### CODE REVIEW

**Comment by willkg** [comment]
Awesome! I want to figure out a test plan before we land this. I'll do that today.

**Comment by willkg** [comment]
@peterbe, @adngdb: Does this look like a good test plan?

Test plan for landing this:
     
1. land the PR
2. wait for -stage to deploy
3. visit the following pages and make sure things look ok:
   1. https://crash-stats.allizom.org/home/product/Firefox
   2. https://crash-stats.allizom.org/topcrashers/?product=Firefox&version=57.0a1&days=7
   3. https://crash-stats.allizom.org/signature/?product=Firefox&version=57.0a1&signature=mozilla%3A%3Adom%3A%3ASelection%3A%3AGetPrimaryFrameForFocusNode&date=%3E%3D2017-08-21T18%3A36%3A26.000Z&date=%3C2017-08-28T18%3A36%3A26.000Z&_columns=date&_columns=product&_columns=version&_columns=build_id&_columns=platform&_columns=reason&_columns=address&_columns=install_time&_sort=-date&page=1#summary (and its tabs)
   4. https://crash-stats.allizom.org/crashes-per-day/?p=Firefox
   5. https://crash-stats.allizom.org/search/?product=Firefox&version=57.0a1&date=%3E%3D2017-08-21T18%3A37%3A00.000Z&date=%3C2017-08-28T18%3A37%3A00.000Z&_sort=-date&_facets=signature&_columns=date&_columns=signature&_columns=product&_columns=version&_columns=build_id&_columns=platform#facet-signature
     
That tests Super Search pretty well.
     
Then we want to compare the old Super Search fields with the new one.

```     
$ wget https://crash-stats.allizom.org/api/SuperSearchFields/
$ cat < index.html | jq -S . > super_search_fields_stage.json
```
     
Then compare the output of that with the attachment super_search_fields_stage.json in the bug. There should be some changes in the descriptions, but otherwise it should be the same.
     
That should cover it.

If that looks ok, I'll copy that to the bug, land this, and proceed accordingly.

**Comment by peterbe** [comment]
I added a thing about looking at crashes with `AccessabilityClient` and `AccessabilityInProcClient` since these were the most recent additions to super search fields.

**Comment by willkg** [comment]
Oops--I missed the "don't cache stuff" commit. I'll do that, too.

**Comment by willkg** [comment]
^^^ That pulls in the "no cache!" changes. Sorry I forgot about those.

**Comment by peterbe** [comment]
And i'm sorry for missing.

**Comment by willkg** [comment]
Landing this now and then going through the test plan after it gets deployed.

File: setup.py [hunk 1]
```diff
@@ -48,6 +48,7 @@ def read(fname):
     test_suite='nose.collector',
     zip_safe=False,
     data_files=[
+        ('socorro/external/es/data', glob.glob('socorro/external/es/data/*.json')),
```
Comment (adngdb) [inline setup.py]: This is not needed, you removed that file.

File: webapp-django/crashstats/supersearch/models.py [hunk 2]
```diff
@@ -33,6 +33,22 @@
...
 
+class SuperSearchFieldsWithoutConfig(super_search_fields.SuperSearchFields):
+    """In the SuperSearchFields implementation class we know with confidence
+    we can just call
+    socorro.external.es.super_search_fields.SuperSearchFields.get() without
+    first having instanciated the base class with the necessary configuration
+    needed to reach Elasticsearch.
+
+    This way we can call `.get()` on an instance of this class, which
+    just reads a .json file.
```
Comment (adngdb) [inline webapp-django/crashstats/supersearch/models.py]: This comment is obsolete.

File: webapp-django/crashstats/supersearch/models.py [hunk 3]
```diff
@@ -216,14 +232,15 @@ def get(self, **kwargs):
 
 class SuperSearchFields(ESSocorroMiddleware):
 
-    implementation = super_search_fields.SuperSearchFields
+    # Read it in once as a class attribute since it'll never change
+    # unless the .json file on disk changes and if that happens you
+    # will have reloaded the Python process.
+    _fields = SuperSearchFieldsWithoutConfig().get()
```
Comment (adngdb) [inline webapp-django/crashstats/supersearch/models.py]: This here is a class attribute, only instanced once. That means if anything modifies it (namely, the ``search_fields`` view in ``supersearch``), every subsequent call of ``get`` will receive the modified version.

File: webapp-django/crashstats/supersearch/models.py [hunk 4]
```diff
@@ -216,14 +232,15 @@ def get(self, **kwargs):
...
 
     API_WHITELIST = None
 
-    # The only reason this data will change is if a user changes it via the UI.
-    # If that happens, the cache will be reset automatically. We can thus
-    # increase the cache a lot here.
-    cache_seconds = 60 * 60 * 24  # 24 hours
+    def get(self):
+        return self._fields
```
Comment (adngdb) [inline webapp-django/crashstats/supersearch/models.py]: How expensive would it be to return a deep copy of that fields object here instead?

**Pull request merged**